src/sources/http_server.rs

```diff
@@ -122,11 +122,60 @@ pub struct SimpleHttpConfig {
     #[configurable(derived)]
     #[serde(default, deserialize_with = "bool_or_struct")]
     acknowledgements: SourceAcknowledgementsConfig,
+
+    /// The namespace to use for logs. This overrides the global setting.
+    #[configurable(metadata(docs::hidden))]
+    #[serde(default)]
+    pub log_namespace: Option<bool>,
 }
 
-impl GenerateConfig for SimpleHttpConfig {
-    fn generate_config() -> toml::Value {
-        toml::Value::try_from(Self {
+impl SimpleHttpConfig {
+    /// Builds the `schema::Definition` for this source using the provided `LogNamespace`.
+    fn schema_definition(&self, log_namespace: LogNamespace) -> Definition {
+        let mut schema_definition = self
+            .decoding
+            .as_ref()
+            .unwrap_or(&default_decoding())
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                SimpleHttpConfig::NAME,
+                parse_value_path(&self.path_key)
+                    .ok()
+                    .map(LegacyKey::Overwrite),
+                &owned_value_path!("path"),
+                Kind::bytes(),
```

Comment:
🤔 if the request is to the root route, do we just insert `%http_server.path."/"`?

Reply:
Looking at the tests, the answer is yes 😄